Release v2

thomas-fred opened the pull request:
Merge features from J-SRAT phase 3 back into `main` as v2.

thomas-fred committed:
Invoke transport failure analysis

thomas-fred committed:
Protect intermediate outputs of criticality assessment

thomas-fred committed:
WIP: single_point_failure_road_rail

thomas-fred committed:
Process one and only one layer per invocation

When processing road nodes, edges were being output as nodes

thomas-fred committed:
Use correct network layers file

thomas-fred committed:
Add bridges as subset of road nodes

Use bridges derived from OSM data
Use from node of bridge edge as bridge node
Factor out some parameters to config

thomas-fred committed:
Remove old version of rules

thomas-fred committed:
Merge pull request #42 from nismod/fix/incorporate_bridges

fix/incorporate_bridges

thomas-fred committed:
Merge branch 'development' into feature/transport_losses

thomas-fred committed:
Restore deleted script

thomas-fred committed:
Tidy script imports

eatyourgreens committed:
feat: benefit cost ratios (#43)

* feat: benefit cost ratios

A python script to calculate the `benefit_cost_ratio` rule.
```sh
python workflow/5_adaptation/benefit_cost_ratio_estimations.py \
 --asset-data processed_data/networks/network_layers_hazard_intersections_details.csv \
 --cost-file results/adaptation_costs/flooding_costs/waste_water_facilities_NWC_nodes_adaptation_timeseries_and_npvs.csv \
 --hazard-label flooding \
 --asset-gpkg waste_water_facilities_NWC \
 --asset-layer nodes \
 --output-dir results
```

* fix: duplicate short label for --cost-file

thomas-fred committed:
Factor out nominal flow data reader

thomas-fred committed:
WIP: Add rail stations failure script & rule

thomas-fred committed:
Merge branch 'development' into feature/single_link_failures

thomas-fred committed:
Fix syntax error

thomas-fred committed:
Merge pull request #45 from nismod/feature/single_link_failures

feature/single_link_failures

thomas-fred committed:
Merge branch 'development' into feature/transport_losses

thomas-fred committed:
Add rule calculating economic impact of rail station closure

thomas-fred committed:
Add a rule for economic losses due to bridge failure

thomas-fred committed:
Catch error on removing missing edge

thomas-fred committed:
Write per-trade sector networks, flow and indicies to disk

Previously we had a bug and were only writing the last sector

thomas-fred committed:
Include node along with edges (becomes label)

thomas-fred committed:
Use OUTPUT and DATA over wildcards

thomas-fred committed:
Remove unused code, apply black

thomas-fred committed:
Factor out loss calculation function

eatyourgreens committed:
feat: adaptation options costs (#50)

* feat: adaptation options costs

A python script to run the `adaptation_options_costs` rule.

```sh
snakemake -c1 results/adaptation_costs/flooding_costs/waste_water_facilities_NWC_nodes_adaptation_timeseries_and_npvs.csv
```

```sh
python workflow/5_adaptation/adaptation_options_costs.py \
 --asset-data workflow/network_layers_hazard_intersections_details.csv \
 --asset-file processed_data/networks/water/waste_water_facilities_NWC.gpkg \
 --cost-file processed_data/adaptation/adaptation_options_and_costs_jamaica.xlsx \
 --hazard-label flooding \
 --asset-gpkg waste_water_facilities_NWC \
 --asset-layer nodes \
 --output-dir results
```

* refactor constants as optional params

eatyourgreens committed:
feat: adaptation damage loss timeseries and npv (#47)

* feat: adaptation damage loss timeseries and npv

A python script to run the `damage_loss_timeseries_and_NPV` rule.
```
snakemake -c1 results/flood_threshold_1p0/loss_damage_npvs/waste_water_facilities_NWC_nodes_EAD_EAEL_npvs.csv
```

```
python workflow/5_adaptation/damage_loss_timeseries_and_npv.py \
 --asset-data workflow/network_layers_hazard_intersections_details.csv \
 --growth-rates-xls processed_data/macroeconomic_data/gdp_growth_rates.xlsx \
 --asset-gpkg waste_water_facilities_NWC \
 --asset-layer nodes \
 --protection-type flood_threshold \
 --threshold 1p5 \
 --output-dir results
```

* add timeseries outputs

* refactor data directory name

* make --data-dir optional

* remove the asset_info loop

* generalise output paths to allow for results/loss_damage_npvs

* PR review suggestions

eatyourgreens committed:
refactor: adaptation bcr calculations (#51)

- fix flake8 warnings
- remove the `asset_info` loop (there should only be one asset row for a given `gpkg` and `layer`.)
- log output file paths, for consistency.

thomas-fred committed:
Merge branch 'development' into feature/transport_losses

eatyourgreens committed:
fix: typo in _adaptation.smk (#52)

thomas-fred committed:
Merge branch 'development' into feature/transport_nodes_failure_analysis

thomas-fred committed:
Reduce chunk count

thomas-fred committed:
Merge pull request #53 from nismod/feature/transport_nodes_failure_analysis

feature/transport_nodes_failure_analysis

thomas-fred committed:
Run black

thomas-fred committed:
Fix failure .csv paths, add logging

thomas-fred committed:
Increase line length to 140

eatyourgreens committed:
refactor: no adaptation bcr input files (#54)

Add `results/loss_damage_npvs` to the list of risk inputs for the adaptation bcr rule. It's used by the Python script, but wasn't listed in the rule dependencies.

thomas-fred committed:
Remove unused functions

eatyourgreens committed:
refactor: adaptation constants (#57)

Move adaptation workflow constants into `config.adaptation_options`.

thomas-fred committed:
Produce labour_trips_and_activity.gpq

Method for producing this file (.gpkg equivalent) was in v1.0, commented
out code. Updated labour_to_work_flow_mapping.py to create it.

thomas-fred committed:
Merge pull request #55 from nismod/feature/single_point_failure_road_rail

- Generate labour_trips_and_activity.gpq
- Implement single_point_failure_road_rail

eatyourgreens committed:
fix: typo in config file (#58)

Fix a typo that broke adaptation rules using `epsg_jamaica` as an input.

eatyourgreens committed:
fix: adaptation options typo (#60)

eatyourgreens committed:
fix: single failure scenario file paths (#59)

I think the sfs file path is `results/economic_losses` not `results/{hazard}_{threshold}/economic_losses`.

eatyourgreens committed:
feat: adaptation rulegraph calculations (#56)

Add adaptation BCR targets to `check_targets.py`.

thomas-fred committed:
Update road network classification and costs (#62)

Update to newer scipy API

eatyourgreens committed:
feat: snakemake adaptation targets (#63)

Add `all_adaptations` rule to Snakefile.

thomas-fred committed:
feature/loss_summary (#64)

* Format with black; update import

* Don't draw DAG, only rulegraph

* Add test framework for helper module

Add a check for numericness of DataFrames

* Various small improvements
- Update to more recent pandas API
- Fix typos
- Check input file extensions

eatyourgreens committed:
refactor: check required file inputs for adaptation bcr (#65)

Raise a visible error in the `benefit_cost_ratio` rule if required input files don't exist. Previously, the file silently exited without a message.

eatyourgreens committed:
fix: consistent naming for asset data (#66)

Use `asset_data` for asset data throughout the adaptation rules.

eatyourgreens committed:
refactor: rename asset_data as network_csv (#67)

Rename `asset_data` to `network_csv` for consistency with other snakemake rules.

eatyourgreens committed:
Revert "fix: consistent naming for asset data (#66)" (#68)

This reverts commit 34a283319fa45b00b11819623dc71b96c73d4c56.

thomas-fred committed:
Wireframe transport hotspots rules (#70)

tomalrussell committed:
Fix minor setup issues (#71)

* Fix snakemake complaint about \d character in wildcard_constraints

* Add graphviz to environment, required by check_targets.py

* Fix README example output file extension

thomas-fred committed:
feat: Exposed value of assets (#72)

* Add hotspots grid metadata file

* Add mean damage cost for water assets

* Remove unused import

* Calculate exposed value of assets by class and sector

thomas-fred committed:
Use consistent naming scheme for aggregations (#69)

Previously, damage_loss_timeseries_and_npv.py was looking for amin
columns that weren't present in EAD/EAEL summary files.

N.B. User must also amend
processed_data/macroeconomic_data/gdp_growth_rates.xlsx to use the
min/max scheme.

thomas-fred committed:
Reorder network layers; add starting IDs for ETL

thomas-fred committed:
Remove last usage of load_config

thomas-fred committed:
Remove dead code

thomas-fred committed:
New module for adaptation functions

thomas-fred committed:
Direct import from scipy

Previously, imported via another module

thomas-fred committed:
Merge raster & transform modules

thomas-fred committed:
Relocate function to calling script

thomas-fred committed:
Move module analysis.utils to top level

thomas-fred committed:
Merge pull request #73 from nismod/refactor-package

refactor: Helper package

thomas-fred committed:
Allow to test for NaN values

thomas-fred committed:
Merge pull request #74 from nismod/is-sole-value-nan

Allow to test for NaN values

thomas-fred committed:
Add unique IDs to assets; save risks as parquet

thomas-fred committed:
Merge pull request #75 from nismod/add-uids

feat: Prepare for visualisation

thomas-fred committed:
Fix mapping to asset_type (used for damage curve lookup)

thomas-fred committed:
Factor out paths

thomas-fred committed:
Merge pull request #76 from nismod/fix-road-network-creation

fix: Road network asset_type tagging

thomas-fred committed:
Employ YAML config to allow comments

thomas-fred committed:
Merge pull request #78 from nismod/yaml-config

feat: Employ YAML config

thomas-fred committed:
Ensure RCP is read from CSV as string

thomas-fred committed:
Reorder rules in file (execution order)

thomas-fred committed:
Ensure RCP is read from CSV as string

thomas-fred committed:
Fix function call; extend logging

thomas-fred committed:
Merge pull request #82 from nismod/adaptation-options

fix: Adaptation options

thomas-fred committed:
Minor changes to documentation

thomas-fred committed:
Merge pull request #83 from nismod/documentation-tweaks

Minor changes to documentation

thomas-fred committed:
Economic loss for transport hotspots

- Remove road and rail edges within hotspots grid cell
- Identify economic flows (labour & trade) which cannot be routed
- Identify cost of rerouting obstructed economic flows
- Output the sum of isolation and rerouting losses on hotspots grid

thomas-fred committed:
Add Gaussian kernel smoothing to hotspots

thomas-fred committed:
Merge pull request #84 from nismod/transport-hotspots

Economic loss for transport hotspots

AdamTaylor9 committed:
initial commit

thomas-fred committed:
Add TODO

thomas-fred committed:
Use correct test invocation

thomas-fred committed:
Remove redundant rule and script

thomas-fred committed:
Change column naming

When running this script for flooding to road nodes (bridges) the output
column names differed from the reference output located on the cluster
(_risk vs _EAD) respectively.

I think the previous implementation never changed the column names as
the str.replace call didn't match anything given the provided input
data.

thomas-fred committed:
Revert to amin/amax naming scheme

thomas-fred committed:
Merge pull request #87 from nismod/fix-adaptation-column-naming

Fix adaptation column naming

AdamTaylor9 committed:
add method to properly assign hazard_threshold for direct_damage rule

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

thomas-fred committed:
Increase number of direct damage ensemble members

thomas-fred committed:
No longer mark file as temporary

thomas-fred committed:
Supply damage curve modification parameters by CSV

Previously, the CSV was used for the nominal case, but not for other
cases (involving adaptation measures).

thomas-fred committed:
Terser path generation with expand

thomas-fred committed:
Factor out adaptation parameters to config

thomas-fred committed:
Sort rules into execution order

thomas-fred committed:
Rasterise once per layer

Previously, each adaptation option permutation would re-rasterise.

AdamTaylor9 committed:
change naming convention of flood protection to asset dictionary parquets

thomas-fred committed:
Fix shell block formatting

Previously, we were missing a set of curly braces and this was flagging
the rule as changed after every run.

thomas-fred committed:
Fix typo in argument name

thomas-fred committed:
Merge pull request #88 from nismod/adaptation-parameters

feat: Adaptation parameters

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

AdamTaylor9 committed:
modify benefit cost ratio to process coastal adaptation case

AdamTaylor9 committed:
modify coastal adaptation bcr to only use threshold values for projected end year date

AdamTaylor9 committed:
create adapt_cost_npv columns for each rcp for coastal adaptation bcr

thomas-fred committed:
Set nodata value for non-positive values

thomas-fred committed:
Merge pull request #89 from nismod/hotspots-kde-set-nodata

Set nodata value for non-positive values

AdamTaylor9 committed:
created new rule to handle generation of combined coastal protection gpkg

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

AdamTaylor9 committed:
modify count_coastal_assets script to produce one csv file instead with inclusion of asset costs

thomas-fred committed:
Ensure bridges are tagged with their linear cost

thomas-fred committed:
Update bridge cost estimates

thomas-fred committed:
Merge pull request #91 from nismod/road-rehab-costs

Road rehab costs

AdamTaylor9 committed:
added coastline protection length to coastline protection assets breakdown

thomas-fred committed:
Update hotspots generation parameters

thomas-fred committed:
Merge pull request #92 from nismod/hotspots-grid-config

Update hotspots generation parameters

thomas-fred committed:
Set bridge edge rehab costs to zero

The wider analysis represents bridges as nodes. Here, the from_node of
the bridge edge has the rehab cost. We ensure it is only set in one
place to reduce the risk of double counting damages.

thomas-fred committed:
Merge pull request #93 from nismod/stop-bridge-double-counting

Set bridge edge rehab costs to zero

thomas-fred committed:
Assume single carriageway when missing data

AdamTaylor9 committed:
modified adaptation_options_costs to handle coastal adaptation case

AdamTaylor9 committed:
added asset cost to asset to coastal protection mappings

AdamTaylor9 committed:
fixed errors in direct damage calcs and asset to coastal protection mapping

tomalrussell committed:
Capitalise short flag for --asset-gpkg to avoid duplicate -g

tomalrussell committed:
Report expected/actual values if assertion fails in cost units check

AdamTaylor9 committed:
filtered out adaptation costs for assets not in the flood protection areas

AdamTaylor9 committed:
modified bcr estimation script  to process coastal adaptation case

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

AdamTaylor9 committed:
modifed coastal adaptation feature generation to only generate for a set rcp, epoch & rp

AdamTaylor9 committed:
modified coastal adaptation rules to work with set rcp, epoch and rp

thomas-fred committed:
Merge pull request #94 from nismod/lane-number-assumption

Assume single carriageway when missing data

AdamTaylor9 committed:
fixed direct damages to work with new coastal protection results format

AdamTaylor9 committed:
check asset max cost column or min cost column if mean cost column not available

AdamTaylor9 committed:
modified benefit cost ratio to work with new coastal protection results format

AdamTaylor9 committed:
moved the multiplication of asset npvs from bcr script to adaptation options cost script

AdamTaylor9 committed:
bug fixes in bcr script

tomalrussell committed:
Sum rasters to use nansum (avoid propagating nans) and pass on nodata value

tomalrussell committed:
Factor out writing indexed splits to tiff

tomalrussell committed:
Avoid duplicate -g short flag with --asset-gpkg-label

tomalrussell committed:
Pass cell_index_* columns through damage calculations

The output is already at the level of split geometries, but didn't contain geometry or cell indices - pass indices through to allow for aggregating back to a grid.

tomalrussell committed:
Avoid duplicate -g short flag with --asset-gpkg-label

tomalrussell committed:
Include baseline hazard layers in hotspots_layers.csv, for direct damage calculations

tomalrussell committed:
Set up a single (central) sensitivity parameter choice for hotspots damage calculations

Both parameters vary from 0-1, used to interpolate between min and max values, so 0.5 is a linear "mean" central estimate.

tomalrussell committed:
Direct damage EAD hotspots, summed various ways and smoothed

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

AdamTaylor9 committed:
modified generate_coastal_adaptations.py to fix edge case errors

thomas-fred committed:
Undo the work of Titivillus

thomas-fred committed:
Fix the cost/damage perturbation where there is no sensitivty ensemble

Previously, I had wrongly assumed 1.0 would be the central estimate.

thomas-fred committed:
Merge pull request #97 from nismod/fix/no-sensitivity

Fix the cost/damage perturbation where there is no sensitivty ensemble

thomas-fred committed:
Merge branch 'development' into feature/hotspot-ead

tomalrussell committed:
Default zero for hotspots rasters

thomas-fred committed:
Merge pull request #95 from nismod/feature/hotspot-ead

Add EAD (direct damages) hotspots

AdamTaylor9 committed:
add new snakemake rule to process coastal adaptation results for visualisation

AdamTaylor9 committed:
Merge branch 'development' into coastal_adaptation_integration

thomas-fred committed:
Add expand rule to run coastal adaptation for all layers

thomas-fred committed:
Add explicit gdal dependency with recent version

Fixes missing proj DB error from splits script

thomas-fred committed:
Remove module level constant

thomas-fred committed:
Do not skip buildings

thomas-fred committed:
Add target rule for coastal protection BCR

thomas-fred committed:
Short circuit when no matching adaptation options

thomas-fred committed:
Handle NoneType flood ID

thomas-fred committed:
Actually create flag file

thomas-fred committed:
Gracefully handle case of no adaptation options

thomas-fred committed:
Gracefully handle empty single failure scenario file

thomas-fred committed:
Add empty single failure scenario file (header only)

thomas-fred committed:
Add coastal adaptation example invocation

thomas-fred committed:
Tidy up -- remove unused imports, formatting

thomas-fred committed:
Reorder expressions for clarity

thomas-fred committed:
Improve readability of adaptation_options_costs function

Rename variables, reorder, reformat

thomas-fred committed:
Don't filter network layers by description

Instead, rely on asset_gpkg and asset_layer to filter network layers. If
we don't have any adaptation options for the assets in that layer, find
that out before short-circuiting.

thomas-fred committed:
Remove variable

thomas-fred committed:
Use original file name

thomas-fred committed:
Skip if (hazard) asset type lookup column is 'none'

thomas-fred committed:
Ignore sbatch jobfiles

thomas-fred committed:
Hardcode path for coastal protection visualisation rule

thomas-fred committed:
Use single line for shell command

Avoid issue with multi-line strings being truncated at line continuation

thomas-fred committed:
Refactor to reduce memory consumption

thomas-fred committed:
Fix logging error

thomas-fred committed:
Update shell blocks to use format rather than f-strings

There seems to be an issue with mixing Python variable interpolation ({DATA}),
literal strings ('areas') and snakemake template variables ({{input.script}})
within an f-string shell block. Using the format method on a string will
circumvent this.

thomas-fred committed:
Add a buildings specific preprocess rule

thomas-fred committed:
Fix line breaks

thomas-fred committed:
Format with black

thomas-fred committed:
Short circuit if no adaptation options available

thomas-fred committed:
Format with black

thomas-fred committed:
Factor out coastal protection to separate script

Find costs to assets avoided by construction of coastal defence. Save these
data in the file typically used for EAD and EAEL.

thomas-fred committed:
Add TODO

thomas-fred committed:
Use (avoided) damage files parameterised by epoch

thomas-fred committed:
Save more data to disk

- Include an `asset_type` field for the area layer
- Save the coastal defence area layer to both output files

thomas-fred committed:
Change edges -> areas

thomas-fred committed:
Split BCR into normal & coastal, factor out functions

thomas-fred committed:
Fix indent level error

thomas-fred committed:
Update docstrings

thomas-fred committed:
Constrain hazard wildcard

thomas-fred committed:
Use correct type (list[str]) for hazards

thomas-fred committed:
Fix typos

thomas-fred committed:
Find BCR for entire coastal protection zones

tomalrussell committed:
Ignore TIFF data files

thomas-fred committed:
Find agg. avoided risk by coastal protection per RP

thomas-fred committed:
Extract coastal protection to own directory; tidy

thomas-fred committed:
Apply black

thomas-fred committed:
Rename ETL scripts

thomas-fred committed:
Remove redundant placeholder rules

thomas-fred committed:
Remove redundant changes to risk calculation

Previously, there were some changes to the `direct_damages` rule and associated
script which aimed to calculate direct damages given a coastal adaptation
option (e.g. sea wall). These are redundant, as downstream we assume all
protected assets have all of their risks averted -- the flood defences are
sized to exclude the maximum flood present in the coastal rasters.

thomas-fred committed:
Use correct argument name and path

tomalrussell committed:
Find upstream catchments of flooded assets

Co-authored-by: Tom Russell <tom.russell@ouce.ox.ac.uk>
Co-authored-by: Robyn Haggis <robyn.haggis@ouce.ox.ac.uk>

tomalrussell committed:
Calculate NbS river flood damage reduction

- rasterise landuse to representative points with forest/afforestable equivalent cover
- intersect with all upstream catchments to calculate peak flow reduction and RP change
- join with flood damages to calculate 'with-nbs' damage, avoided damages
- accumulate split avoided damages to the flood grid
- WIP estimate of afforestation damage reduction effect, on the landuse grid / catchment scale

Co-authored-by: Tom Russell <tom.russell@ouce.ox.ac.uk>
Co-authored-by: Robyn Haggis <robyn.haggis@ouce.ox.ac.uk>

tomalrussell committed:
Fix RP damage bug, aggregate avoided damages to damage location / afforestable locations

tomalrussell committed:
Allow for fast smoothing if KDE output resolution matches input

tomalrussell committed:
Smooth NbS river damage/avoided/reduction tiffs

tomalrussell committed:
Coarsen and smooth avoided damages TIFFs

thomas-fred committed:
Merge pull request #86 from nismod/coastal_adaptation_integration

Coastal Adaptation

tomalrussell committed:
Check total avoided damages values in different formats

tomalrussell committed:
Merge branch 'development' into feature/nbs-catchments-dev

tomalrussell committed:
Create paper directory for analysis/visualisation notebooks

tomalrussell committed:
Merge pull request #99 from nismod/feature/nbs-catchments-dev

NbS river flood catchments method

tomalrussell committed:
Merge branch 'main' into development

tomalrussell commented on Dockerfile:
This looks incomplete and is not mentioned in any dev docs - suggest dropping it entirely. Devcontainer or singularity might be better solutions depending on the goals.

thomas-fred committed:
Add documentation on inputs and outputs

thomas-fred committed:
Remove newlines after headings

thomas-fred committed:
Amend data documentation

tomalrussell committed:
Add adapted methodology report as markdown

tomalrussell committed:
Fix typos and some table references

tomalrussell committed:
Fix some latex and formatting

tomalrussell committed:
Try fixing latex/markdown parsing as italics

tomalrussell committed:
Fix more latex/markdown parsing

tomalrussell committed:
Merge pull request #102 from nismod/data-documentation

Add documentation on inputs and outputs

tomalrussell commented on workflow/1_damage/_damage.smk:
```diff
@@ -0,0 +1,174 @@
+What are the costs associated with rebuilding damaged assets?
+"""
+
+rule write_hazard_transforms:
```
I don't think this is used either - can drop

tomalrussell commented on workflow/1_damage/split_networks.py:
```diff
@@ -0,0 +1,330 @@
+    type=click.Path(exists=False, dir_okay=False, file_okay=True, readable=True),
+    help="Path to save splits as geoparquet",
+)
+def split_assets(network_csv, hazard_csv, data_dir, asset_gpkg, asset_layer, output_path):
```
Close overlap with `snail split` CLI.. if it's not a straight swap, suggest logging an issue, might be a "good first issue" or one for the backlog.

tomalrussell commented on workflow/2a_transport/_transport.smk:
```diff
@@ -0,0 +1,252 @@
+        network = snkit.network.add_topology(network, id_col="id")
+
+        logging.info("Label bridge nodes (from_id of bridge edge)")
+        # previously, bridge data from the NWA (2016) was used
```
Update docs to record use of OSM highway and bridge data, and table mapping classifications.

tomalrussell commented on workflow/3_criticality/rail_stations_failure_analysis.py:
```diff
@@ -0,0 +1,115 @@
+)
+
+tqdm.pandas()
+epsg_jamaica = 3448
```
This is quite a common declaration - does something like `from jamaica_infrastructure.constants import epsg_jamaica` sound any better to your ears?

tomalrussell commented on workflow/3_criticality/rail_stations_failure_analysis.py:
```diff
@@ -0,0 +1,115 @@
+    from multi-modal transport network.
+    """
+
+    # 0.6 - 2.1% of the value per day
```
Cross-link to the docs https://github.com/nismod/jamaica-infrastructure/blob/b7dd6d82d50853b7e4d75f69ba669abd6c91ed7f/docs/appendix-c-network-flow-models.md?plain=1#L554

tomalrussell commented on workflow/3_criticality/roads_bridges_failure_analysis.py:
```diff
@@ -0,0 +1,116 @@
+    from multi-modal transport network.
+    """
+
+    # 0.6 - 2.1% of the value per day
```
Can we avoid stating this assumption twice? (also in `rail_stations_failure_analysis`)

tomalrussell commented on workflow/3_criticality/single_link_failures.py:
```diff
@@ -0,0 +1,122 @@
+    Calculate economic costs of removing road edges from multi-modal transport network.
+    """
+
+    # 0.6 - 2.1% of the value per day
```
Wage assumption

tomalrussell commented on workflow/3_criticality/transport_single_point_failure_results_combine.py:
```diff
@@ -0,0 +1,219 @@
+)
+def main(*, processed_data_dir, results_dir):
+
+    hourly_wage = 0.4 * (1 + 0.454) * 235.25  # Between 200 - 500 JMD for 2012 stats, 45.4% inflation in currency
```
Wage assumption

tomalrussell commented on workflow/4b_hotspots/_hotspots.smk:
```diff
@@ -0,0 +1,440 @@
+"""
+The rules in this file produce raster maps of 'hotspots' of infrastructure risk.
+
+1) Define a raster hotspots grid
```
Add hotspots method section to docs

tomalrussell commented on workflow/5b_coastal_adaptation/1_generate_defence_zones.py:
```diff
@@ -0,0 +1,2357 @@
+    if not isinstance(coast_segment, (LineString, MultiLineString)) or coast_segment.length == 0:
+        return []
+
+    # t_scale = 10_000_000
```
Drop unused code

tomalrussell commented on workflow/5b_coastal_adaptation/1_generate_defence_zones.py:
```diff
@@ -0,0 +1,2357 @@
+    return geopkg  # Return the updated GeoDataFrame with the new column
+
+
+""" Generating Flood Zones Polygons via S-CAPE Method
```
Add coastal zone adaptation method note to `docs`

tomalrussell committed:
Lint coastal adaptation scripts

- snake_case function names
- sort imports and remove unused
- format with black

tomalrussell commented on workflow/utilities/_utilities.smk:
```diff
@@ -0,0 +1,31 @@
+    filepath = checkpoints.sensitivity_parameters.get(**wildcards).output.sensitivity_parameters
+    return len(pd.read_csv(filepath))
+
+def sensitivity_id_from_slug(wildcards):
```
Might as well:
```suggestion
def sensitivity_id_from_slug(wildcards) -> str:
```

tomalrussell commented on workflow/utilities/check_workflow.py:
```diff
@@ -0,0 +1,87 @@
+
+required_files = set()
+for f, cols in csv_path_refs.items():
+    csv_file = pandas.read_csv(f"https://github.com/nismod/irv-jamaica/raw/refs/heads/main/etl/{f}.csv")
```
Consider pinning to a tagged release of `irv-jamaica`?

```suggestion
    TAG="0.4.29"
    csv_file = pandas.read_csv(f"https://github.com/nismod/irv-jamaica/raw/refs/tags/{TAG}/etl/{f}.csv")
```

The release/update/check process here goes something like:
- do work in `jamaica-infrastructure` to produce updated outputs
- update `irv-jamaica` to handle any new layers if necessary
- update this check as a high-level regression test of the workflow outputs

thomas-fred replied:
Yes, that makes sense to me.

This script checks if output files exist. The other one checks if there's a route through the workflow to create them. They should probably swap names given their contents. I'd have the one that checks if a route through the workflow is possible use the `irv-jamaica` file lists as input. Then perhaps do without the one that checks for files on disk -- that's what `snakemake` does.

tomalrussell reviewed (changes requested):
Big PR! Great to have this under the Snakemake workflow.

Should be minor queries/suggestions in comments - anything that would block too long can be pushed to an issue.

tomalrussell committed:
Move economic land use and buildings preprocess scripts to folder

thomas-fred committed:
Remove embryonic Dockerfile

thomas-fred committed:
Remove redundant rule

thomas-fred committed:
Update documentation of road network generation

thomas-fred committed:
Add library constant for local CRS code

thomas-fred committed:
Pass wage and trade disruption as arguments

thomas-fred committed:
Update workflow/utilities/_utilities.smk

Co-authored-by: Tom Russell <tom.russell@ouce.ox.ac.uk>

thomas-fred committed:
Add hotspots docs

thomas-fred committed:
Remove redundant code, comments

thomas-fred committed:
Add coastal protection method docs

thomas-fred committed:
Escape line break in f-strings

thomas-fred committed:
Escape for line breaks in multi-line strings

thomas-fred committed:
Remove redundant f-string marker

thomas-fred committed:
Use correct test invocation path

thomas-fred committed:
Use correct path

thomas-fred committed:
Add coastal targets

thomas-fred committed:
Rewrite as async

thomas-fred committed:
Add building files as explicit outputs

thomas-fred committed:
Target specific irv-jamaica release

Co-authored-by: Tom Russell <tom.russell@ouce.ox.ac.uk>

thomas-fred committed:
Rename scripts to better reflect contents

thomas-fred committed:
List new appendices on docs front page
